packages/compiler/src/compiler.ts

```diff
@@ -273,6 +285,27 @@ export class ComponentCompiler {
       throw new Error(`Component not found at ${componentPath}`);
     }
 
+    const isLocalComponent = this.localComponents.get(componentPath);
+    // In case the component block height has been defined - clean it and then add it from the module entry
+    const [cleanComponentPath] = componentPath.split('@');
+    const componentCacheKey = `${cleanComponentPath}@${moduleEntry?.blockHeight}`;
```

Comment:
Should we omit the `@` suffix entirely if the block height isn't provided? I would expect a `blockHeight` values of `null`, `undefined`, and `""` to evaluate to the same cache key, but interpolating it directly yields different values.

Reply:
Great catch! There should not be a case when the `@` suffix is followed by an empty value. The whole point of this check is to see if the block height has been attached in the import statement and replace it with the actual component block height from the rpc request. Looks like there was a bug in the initial version of this PR - when the feature flag was ignored and the component sources could be cached even if there was no block height, which resulted the `<Component>@undefined` to appear in the indexedDB. I've updated `compiler.ts`, so now this behavior should be impossible.